src/irmin-pack/unix/file_manager.ml

```diff
@@ -736,4 +735,21 @@ struct
     | T15 ->
         (* Unreachable *)
         assert false
+
+  (* TODO : this does not work yet with GC, missing updates to CF fields. *)
```

Comment:
Are you planning to update the control file fields? I think it is independent of GC work so you can do that in this PR (and add tests that verify reopening store after some splits still works).